lib/core/base/virtual-node/serial-virtual-node.js

```diff
@@ -60,6 +94,12 @@ function normaliseProps(serialNode) {
     typeof nodeName === 'string',
     `nodeName has to be a string, got '${nodeName}'`
   );
+  assert(
+    NodeNamesToTypesEnum[nodeName] === nodeType ||
+      !(nodeName in NodeNamesToTypesEnum) ||
+      !(nodeType in [2, 3, 8, 9, 11]),
+    `nodeName and nodeType have to match`
+  );
```

Comment:
I don't think this assert is needed as we can let the previous asserts catch invalid types of nodeType and nodeName. We don't want to test too much into valid values.

```suggestion
```